optional/vtd/vendor/vtd-2022.3/Dockerfile

```diff
@@ -138,10 +136,21 @@ RUN --mount=type=cache,id=ubuntu-20.04-cache,target=/var/cache/apt \
         tcsh \
         xterm \
         wget \
-        && \
-    wget 'https://launchpad.net/~ubuntu-security/+archive/ubuntu/ppa/+build/15108504/+files/libpng12-0_1.2.54-1ubuntu1.1_amd64.deb' -O /root/libpng12.deb && \
-    dpkg -i /root/libpng12.deb && \
-    rm /root/libpng12.deb && \
+        libncursesw5 \
+        libpng16-16 \
+        libwebp6 \
+        libffi7\
+        zlib1g-dev
+        
+RUN wget 'https://ppa.launchpadcontent.net/linuxuprising/libpng12/ubuntu/pool/main/libp/libpng/libpng_1.2.54.orig.tar.xz' && \ 
+    tar Jxfv libpng_1.2.54.orig.tar.xz && \
+    cd libpng-1.2.54 && \
+    ./configure && \
+    make && \
+    make install && \
+    mv /usr/local/lib/libpng12.so.0.54.0 /lib/x86_64-linux-gnu/libpng12.so.0.54.0  && \
+    cd /lib/x86_64-linux-gnu  && \
+    ln -s libpng12.so.0.54.0 libpng12.so.0 && \
```

Comment:
If you are going to do this, then you might as well get libpng from the source:

https://sourceforge.net/projects/libpng/files/libpng12/1.2.59/

And when building, make sure to build in a temporary directory somewhere which you remove afterwards. Then, copy the resulting `.so` file directly to where you want it instead of first running `make install`.

Reply:
applied your suggestion